TFT/src/User/Menu/PrintingMenu.c

```diff
@@ -592,7 +592,7 @@ void menuPrinting(void)
     if (infoMachineSettings.longFilename == ENABLED && infoFile.source == BOARD_SD)
```

Comment:
As it was done in Print.c

in PrintingMenu.c:

```
    if (infoMachineSettings.longFilename == ENABLED && infoFile.source == BOARD_SD)
      printingItems.title.address = (uint8_t *) infoFile.longFile[infoFile.fileIndex];
    else
      printingItems.title.address = (uint8_t *) infoFile.file[infoFile.fileIndex];

```
is equivalent to:

`    printingItems.title.address = hideFileExtension(infoFile.fileIndex);`

in Printing.c

```
  if (infoMachineSettings.longFilename == ENABLED && infoFile.source == BOARD_SD)
    sprintf(infoPrintSummary.name,"%." STRINGIFY(SUMMARY_NAME_LEN) "s", infoFile.longFile[infoFile.fileIndex]);
  else
    sprintf(infoPrintSummary.name,"%." STRINGIFY(SUMMARY_NAME_LEN) "s", infoFile.file[infoFile.fileIndex]);
```

is equivalent to:

`  sprintf(infoPrintSummary.name,"%." STRINGIFY(SUMMARY_NAME_LEN) "s", hideFileExtension(infoFile.fileIndex));
`

As already said, in both cases there is no need to call restoreFileExtension and the original menuDrawTitle can be restored. Also hideExtension and restoreExtension will be no more exposed for public use

Reply:
This suggestion is wrong in many ways. 
- Hiding an extension for an already hidden file name can lead to unpredictable results (ex: "Cover.gembird.gcode"). Added an extra check to not hide extension successively.
- Hiding extension for print summary is the opposite of the intention. It was a consensus to show it.
- Leaving the filename in an unknown state is anything but safe. Better hide it only when needed and restore it so no surprises down the line. (See cause of bug #2300)